sdk/typescript/src/providers/json-rpc-provider.ts

```diff
@@ -502,6 +504,24 @@ export class JsonRpcProvider extends Provider {
     }
   }
 
+  async getTransactionWithAuthSigners(
+    digest: TransactionDigest
+  ): Promise<SuiTransactionWithAuthSignersResponse> {
+    try {
+      const resp = await this.client.requestWithType(
+        'sui_getTransactionWithAuthSigners',
```

Comment:
Why is this a separate RPC method vs a flag that we can call `getTransaction` with?

Reply:
good point, we can use a flag

Reply:
imo we should either
1. have the single `getTransaction` endpoint with a flag as @Jordan-Mysten suggested
2. Add a new `getAuthorityNames` endpoint but only return `Vec<AuthorityName>`. The Explorer will still use the `getTransaction` endpoint and only make a request to the new `getAuthorityNames` endpoint when the user clicks on the `Signature` tab.

I am leaning towards 2 because `getTransaction` is one of the most heavily used endpoints and fetching/including a list of authority names(especially when we have 100+ validators) in the `getTransaction` response is expensive. My hunch is that most of the Explorer users do not click on the `Signature` tab at all.

Reply:
I agree w Chris, let's have a different endpoint.

Reply:
As there many opinions here I leave it to @Jordan-Mysten and @pchrysochoidis to decide. Let's get a decision by tomorrow, so we can unblock Joy.

Reply:
2. Sounds reasonable to me

Reply:
@pchrysochoidis for 2) its a bigger refactor (https://github.com/MystenLabs/sui/pull/6337#issuecomment-1328123646), unless we have more compiling use case for a more modular react components, i would prefer 1) to optionally return the auth signers key to the same getTransaction endpoint.